### PR TITLE
Add VSO configs table

### LINKED ISSUE #8960: Add IHP tasks for all PVA cases
PVA can submit IHPs for all docket types (unlike other IHP-writing VSOs who can only submit IHPs for Direct Review and Evidence Submission)

## AC
- IHP tasks are added for all PVA cases, regardless of docket type

### PR DESCRIPTION
**Pull request by lowellrex**
Resolves #8960 and connects #8961. This PR adds the new `vso_configs` table.

After this change is pushed live, we will need to add a row in this table for PVA to complete the requirements in the original ticket. We can do that by running the following in the production rails console:

```ruby
rails c> v = Vso.find_by(name: "Paralyzed Veterans of America")
rails c> conf = VsoConfig.create(organization: v, ihp_dockets: [Constants.AMA_DOCKETS.direct_review, Constants.AMA_DOCKETS.evidence_submission, Constants.AMA_DOCKETS.hearing])
```

Additionally, in order to satisfy part of the requirements for #8961 (there will be more work left to do in order to hide tabs) we can do something similar (or maybe include this in a method on the VSO class?):

```ruby
rails c> v = Vso.find_by(name: "Example Field VSO")
rails c> conf = VsoConfig.create(organization: v, ihp_dockets: [])
```

### CODE REVIEW

File: spec/models/tasks/root_task_spec.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
       context "when it has an ihp-writing vso" do
         let(:appeal) do
-          create(:appeal, docket_type: "direct_docket", claimants: [
+          create(:appeal, docket_type: Constants.AMA_DOCKETS.direct_review, claimants: [
```
Comment (lowellrex) [inline spec/models/tasks/root_task_spec.rb]: These docket types were named incorrectly. Updated them to our new constants, but didn't go through the rest of this file to update docket types to constants in order to keep the size of this PR down. That work will be done in #9492.

File: spec/models/tasks/root_task_spec.rb [hunk 2]
```diff
@@ -131,6 +131,40 @@
         expect(ScheduleHearingTask.find_by(appeal: appeal).parent.class.name).to eq("HearingTask")
         expect(ScheduleHearingTask.find_by(appeal: appeal).parent.parent.class.name).to eq("DistributionTask")
       end
+
+      context "when VSO writes IHPs for hearing docket cases" do
```
Comment (lowellrex) [inline spec/models/tasks/root_task_spec.rb]: Tests for Root Task logic.

File: spec/models/vso_spec.rb [hunk 3]
```diff
@@ -82,4 +82,187 @@
       end
     end
   end
+
+  describe ".should_write_ihp?" do
```
Comment (lowellrex) [inline spec/models/vso_spec.rb]: Tests for Vso and VsoConfig additions.

File: spec/models/vso_spec.rb [hunk 4]
```diff
@@ -82,4 +82,187 @@
...
+        context "when the appeal is on the hearing docket" do
+          let(:docket) { Constants.AMA_DOCKETS.hearing }
+          it "should return false for all docket types" do
+            expect(subject).to eq(false)
+          end
+        end
+      end
+    end
+
+    context "when vso_configs record for this VSO contains all 3 AMA docket types (field VSOs)" do
```
Comment (tomas-nava) [inline spec/models/vso_spec.rb]: should be something like `"when vso_configs record for this VSO doesn't contain any docket types"`

**Pull request merged**